scripts/Muon/GUI/MuonAnalysis/loadrun/load_run_presenter.py

```diff
@@ -0,0 +1,260 @@
+
+        if len(file_names) > 1 and not self._load_multiple_runs:
+            self._view.warning_popup("Multiple files selected in single file mode")
+            self._view.reset_run_edit_from_cache()
+            return
+
+        self.handle_loading(file_names, self._use_threading)
+
+    def handle_loading(self, filenames, threaded=True):
+        if threaded:
+            self.handle_load_thread_start(filenames)
+        else:
+            self.handle_load_no_threading(filenames)
+
+    def handle_load_no_threading(self, filenames):
+        self.on_loading_start()
+        self._model.loadData(filenames)
+        try:
+            self._model.execute()
+        except ValueError as e:
+            self._view.warning_popup(e.args[0])
+        self.on_loading_finished()
+
+    def on_loading_start(self):
+        self._view.notify_loading_started()
+        self.disable_loading()
+
+    def handle_load_thread_start(self, filenames):
+        self.on_loading_start()
+
+        self._load_thread = self.create_load_thread()
+        self._load_thread.threadWrapperSetUp(self.disable_loading, self.handle_load_thread_finished)
+        self._load_thread.loadData(filenames)
+        self._load_thread.start()
+
+    def handle_load_thread_finished(self):
+        self._load_thread.threadWrapperTearDown(self.disable_loading, self.handle_load_thread_finished)
+        self._load_thread.deleteLater()
+        self._load_thread = None
+
+        self.on_loading_finished()
+
+    def on_loading_finished(self):
+        # If in single file mode, remove the previous run
+        if not self._load_multiple_runs and len(self._model.loaded_runs) > 1:
+            self._model.remove_previous_data()
+
+        run_list = self._model.loaded_runs
+        self.set_run_edit_from_list(run_list)
+
+        self._view.notify_loading_finished()
+        self.enable_loading()
+
+    # ------------------------------------------------------------------------------------------------------------------
+    # Loading from current run button
+    # ------------------------------------------------------------------------------------------------------------------
+
+    def handle_load_current_run(self):
+
+        try:
+            current_run_file = file_utils.get_current_run_filename(self.get_current_instrument())
+        except ValueError as e:
+            self._view.warning_popup(e.args[0])
+            return
+
+        if current_run_file == "":
+            self._view.warning_popup("Cannot find directory for current instrument : " + self._instrument)
+            return
+
+        self.handle_loading_current_run([current_run_file], self._use_threading)
+
+    def handle_loading_current_run(self, filenames, threaded=True):
+        if threaded:
+            self.handle_load_current_run_thread_start(filenames)
+        else:
+            self.handle_load_current_run_no_threading(filenames)
+
+    def handle_load_current_run_no_threading(self, filenames):
```

Comment:
Did you add the switch for threading to the thread wrapper?

Reply:
Yes, it is in this file and called `self._use_threading`

Reply:
shall we use that instead?

Reply:
We do use that.. ? the `self._use_threading` flag completely determines whether threading is used or not within the presenter.

Reply:
sorry I meant the switch from within the threading model